### DISCUSSION TITLE
Why ButtonInput.pressed() parameter "input" need ownership ?

### DISCUSSION THREAD

**Discussion by Aqaao** [bevyengine/bevy#14180] [General]
https://docs.rs/bevy/0.14.0/bevy/input/struct.ButtonInput.html#method.pressed
```rust
  pub fn pressed(&self, input: T) -> bool {
      self.pressed.contains(&input)
  }

  pub fn any_pressed(&self, inputs: impl IntoIterator<Item = T>) -> bool {
      inputs.into_iter().any(|it| self.pressed(it))
  }
```
Apparently only immutable reference are used in function body.  When I define multiple shortcuts for an action, I need to create `Vec<KeyCode>` to store multiple KeyCode. But call `any_pressed()` need force deep copy Vec, this will incur additional overhead. 

"input" parameter's ownership is necessary?

**Comment by Litttlefish** [comment]
Because `input` type is T instead of &T?

**Comment by mockersf** [comment]
No reason, it can take references: https://github.com/bevyengine/bevy/pull/14182

I think it's for historical reasons, where built in inputs are small and copy and clone, so taking references didn't change anything

**Reply by Aqaao** [reply]
😘
